src/components/tree/tree.e2e.ts

```diff
@@ -904,5 +904,39 @@ describe("calcite-tree", () => {
 
       expect(await getActiveElementId(page)).toEqual("child-1");
     });
+
+    it("renders when tree is the topmost element in a shadow root where it has no parent", async () => {
+      const templateHTML = html`<template id="test-tree-element-template">
+        <calcite-tree>
+          <calcite-tree-item>Child</calcite-tree-item>
+        </calcite-tree>
+      </template>`;
+
+      const page = await newE2EPage();
+      await page.setContent(
+        html`
+          ${templateHTML}
+          <test-tree-element></test-tree-element>
+        `
+      );
+      await page.waitForChanges();
+
+      const defineCustomElement = await page.evaluate(async (): Promise<void> => {
+        customElements.define(
+          "test-tree-element",
+          class extends HTMLElement {
+            connectedCallback() {
+              this.attachShadow({ mode: "open" }).append(
+                (document.getElementById("test-tree-element-template") as HTMLTemplateElement).content.cloneNode(true)
+              );
+            }
+          }
+        );
+      });
+      await page.waitForChanges();
+
+      renders("calcite-tree", { display: "block" });
+      await expect(() => defineCustomElement).not.toThrow(TypeError);
```

Comment:
I _think_ this needs to wait for page changes in the function that goes over the error scenario.

Can you try the following updates:

```ts
const defineTestElementAndWaitForChanges = async () => {
  await page.evaluate(async (): Promise<void> => {
        customElements.define(
          // ...
  });
  await page.waitForChanges();
};

// no `renders` call as this should be captured in `defineTestElementAndWaitForChanges`

await expect(() => defineTestElementAndWaitForChanges).not.toThrow(Error);
```
?

If this fails when you roll back your fix and passes when you restore them. This should be good to go.

Reply:
Yeah, it fails when the wait is moved inside the function. Good point about the render call.